File: backend/src/skpl_agent/app/_router/_schema/_desktop_node.py
"""Request/response schemas for desktop node endpoints."""

from __future__ import annotations

from pydantic import BaseModel


class DesktopNodeResponse(BaseModel):
    """A registered desktop node visible to the frontend."""
    node_id: str
    node_name: str
    status: str  # connecting | online | idle | busy | offline
    os_name: str
    os_version: str
    python_version: str
    screen_width: int
    screen_height: int
    cpu_count: int
    total_memory_mb: int
    capabilities: list[str]
    cpu_percent: float
    memory_percent: float
    active_actions: int
    registered_at: str
    last_seen: str
    is_available: bool


class DesktopNodeListResponse(BaseModel):
    nodes: list[DesktopNodeResponse]
    total: int
    online_count: int

File: frontend/src/api/desktopNode.ts
/**
 * Desktop Node API client.
 *
 * Endpoints:
 * - GET /api/desktop/nodes        — list registered desktop nodes
 * - GET /api/desktop/nodes/stats  — node registry statistics
 */
import { client } from './client';

export interface DesktopNode {
  node_id: string;
  node_name: string;
  status: string;  // connecting | online | idle | busy | offline
  os_name: string;
  os_version: string;
  python_version: string;
  screen_width: number;
  screen_height: number;
  cpu_count: number;
  total_memory_mb: number;
  capabilities: string[];
  cpu_percent: number;
  memory_percent: number;
  active_actions: number;
  registered_at: string;
  last_seen: string;
  is_available: boolean;
}

export interface DesktopNodeListResponse {
  nodes: DesktopNode[];
  total: number;
  online_count: number;
}

export const desktopNodeApi = {
  listNodes: () =>
    client.get<DesktopNodeListResponse>('/api/desktop/nodes'),

  getStats: () =>
    client.get<{
      total_nodes: number;
      online_nodes: number;
      offline_nodes: number;
      available_nodes: number;
      by_tenant: Record<string, number>;
      by_os: Record<string, number>;
    }>('/api/desktop/nodes/stats'),
};